ft: deprecate client mdonly check

ssalaues opened the pull request:
Deprecates the `x-amz-meta-mdonly` header check in the client since this is now handled by a higher order function here:
https://github.com/scality/cloudserver/pull/2025

bert-e commented:
# Hello ssalaues,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/arsenal/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @rachedbenmustapha

* @rahulreddy

* @bennettbuchanan

alexanderchan-scality commented on lib/storage/data/DataWrapper.js:
```diff
@@ -918,7 +920,23 @@ class DataWrapper {
+            }
+            const dataRetrievalInfo = {
+                key: keyContext.objectKey,
+                dataStoreName: this.client.clients[location],
```
```suggestion
                dataStoreName: location,
```

ssalaues replied:
Oops yeah you're right

alexanderchan-scality commented on lib/storage/data/DataWrapper.js:
```diff
@@ -918,7 +920,24 @@ class DataWrapper {
+            const dataRetrievalInfo = {
+                key: keyContext.objectKey,
+                dataStoreName: this.client.clients[location],
+                dataStoreType: this.client.clients[location].clientType,
```
will need to perform a check to make sure `this.client.client` exists.
`file / mem` clients do not have the `client.client` map

ssalaues replied:
Will do, thank you!

Commit:
ft: deprecate mdonly check

This commit deprecates the mdonly check for the PFSclient in favor of a mdonly
check in a higher order function.

https://github.com/scality/cloudserver/pull/2025

ssalaues commented:
Closing this PR as it is no longer needed. The PFSclient will still need the header check because of how the 0 bytes are handled in the higher function.
scality/cloudserver#2025
